source/language/openpulse.rst

```diff
@@ -0,0 +1,609 @@
+here has several advantages over the original JSON format:
+
+- Improved readability
+- Pulse timing is based on instruction ordering and works with programs containing branching
+  control flow
+- Reusable gate calibrations enable more succinct calibration descriptions.
+- Pulse definitions are declared as a calibration for individual circuit instructions attached to
+  physical qubits enabling the microcoding of gate level operations.
+- Richer ability to compose complex pulses through natural DSP-like operations.
+- Clearly defined relationship between pulses, channels and the phases of compilation to
+  hardware resources.
+- Use of multiple frames on a single channel at the same time (see :ref:`Geometric gate`)
+
+
+Openpulse provides a flexible programming model that should extend to many quantum control schemes
+and hardware. At the core of the OpenPulse grammar are the concepts of ``frames``.
+``frames`` are used to schedule the playing of a ``waveform`` or capturing of data
+via a ``channel`` on the target device. A channel is a software abstraction which allows the
+programmer to be agnostic to complexities of the device's underlying pulse generation hardware. It
+is the responsibility of the target device's compiler to map ``frames`` to the applied channels in
+target hardware.
+
+
+Channels
+--------
+
+Channels map to a hardware resource, which can play pulses to manipulate a qubit
+or capture data from the qubit to perform a measurement. There is a many-to-many
+relationship between qubits and channels. One qubit may have multiple channels
+connecting to it. Pulses on different channels would have different physical
+interactions with that qubit. A channel may also have many qubits. For instance,
+a channel could manipulate the coupling between two neighboring qubits, or
+could even reference multiple qubits in a chain.
+
+Channels are vendor-specific and device-specific. It is expected that vendors
+of quantum hardware provide the appropriate channel names and qubit mappings
+as configuration information to end users. Each channel may also have associated
+static settings, such as local-oscillator frequencies, which do not vary
+throughout program execution. Again it is expected that vendors of quantum
+hardware provide a method for manipulating those static settings if appropriate.
+
+There are two kinds of channels: transmit channels (sending input to a quantum
+device) and receive channels (reading output from a quantum device).
+
+A channel is only used to specify the physical resource on which to play a pulse or from which
+to capture data. This specification should be done by providing a mapping between a qubit list +
+name and the configured hardware channel. The hardware can then be accessed as
+OpenPulse ``txchannel``/``rxchannel``'s via "get" functions.
+
+.. code-block:: c
+
+    txch(qubit q0, ..., qubit qn, str name) -> txchannel  // get transmit channel
```

Comment:
Should we add `$` (ie `qubit $q0`) to be explicit here about the physical qubits?

Reply:
We should also change these to `channel` as was decided offline (and as has already been done in some of the examples).

Reply:
Fixed in e5b0181.